src/lib/Repository/Values/Content/UiRelation.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Repository\Values\Content;
+
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\Core\Repository\Values\Content\Relation;
+use eZ\Publish\API\Repository\Values\Content\Relation as APIRelation;
+
+/**
+ * Extends original value object in order to provide additional fields.
+ */
+class UiRelation extends Relation
+{
+    /**
+     * Field definition.
+     *
+     * @var string
+     */
+    protected $fieldDefinition;
```

Comment:
I'd change it to `fieldDefinitionName`. `FieldDefinition` alone has a strong meaning in our domain.